libvast/include/vast/partition_synopsis.hpp

```diff
@@ -90,6 +90,9 @@ struct partition_synopsis final : public caf::ref_counted {
   friend partition_synopsis* ::caf::default_intrusive_cow_ptr_unshare<
     partition_synopsis>(partition_synopsis*& ptr);
   partition_synopsis* copy() const;
+
+  // Cached memory usage.
+  size_t memusage_ = 0ull;
```

Comment:
Probably needs to be made `mutable` now, with a note that the function calculating the memusage is no longer thread safe despite being marked `const`. Or, alternatively, you no longer mark that function `const`.

Reply:
I actually went for both by using an atomic here.